From 476b26f31d4237be3fe61148a8f2390532d526ff Mon Sep 17 00:00:00 2001
From: Madhusudhan S <madhusudhan.s@intel.com>
Date: Thu, 31 Jan 2019 17:56:30 -0500
Subject: [PATCH 3/4] Changing usb config during suspend and resume

Changing the usb config to "none" during S3 and enabling
the usb config to "adb" to resume adb.

Change-Id: Ic445c1df5fbefb55f7d9505fed3a3375931a1cbd
Tracked-on: OAM-71949
Signed-off-by: Madhusudhan S <madhusudhan.s@intel.com>
---
 .../java/com/android/server/power/PowerManagerService.java | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/services/core/java/com/android/server/power/PowerManagerService.java b/services/core/java/com/android/server/power/PowerManagerService.java
index 9eb84db0eb75..30464cfdc7cf 100644
--- a/services/core/java/com/android/server/power/PowerManagerService.java
+++ b/services/core/java/com/android/server/power/PowerManagerService.java
@@ -123,6 +123,7 @@ import android.os.FileUtils;
 public final class PowerManagerService extends SystemService
         implements Watchdog.Monitor {
     private static final String TAG = "PowerManagerService";
+    private static final String USB_CONFIG_PROPERTY = "sys.usb.config";
 
     private static final boolean DEBUG = false;
     private static final boolean DEBUG_SPEW = DEBUG && true;
@@ -1533,6 +1534,8 @@ public final class PowerManagerService extends SystemService
                 default:
                     Slog.i(TAG, "Going to sleep by application request (uid " + uid +")...");
                     reason = PowerManager.GO_TO_SLEEP_REASON_APPLICATION;
+		    SystemProperties.set(USB_CONFIG_PROPERTY, "none");
+
 		    // Adding force suspend code to enable IVI to enter S3 after pressing sleep button
                     try {
                         FileUtils.stringToFile("/sys/power/state", "mem");
@@ -4477,6 +4480,10 @@ public final class PowerManagerService extends SystemService
                 public void run() {
                     final long ident = Binder.clearCallingIdentity();
                     try {
+						// Setting usbconfig to "adb" to enable ADB after wakeup from S3
+						String currUsbConfig = new String(SystemProperties.get(USB_CONFIG_PROPERTY, "none"));
+						if(currUsbConfig.equals("none"))
+							SystemProperties.set(USB_CONFIG_PROPERTY, "adb");
                         wakeUpInternal(eventTime, reason, uid, opPackageName, uid);
                     } finally {
                         Binder.restoreCallingIdentity(ident);
-- 
2.17.1

